libqtile/core/manager.py

```diff
@@ -530,19 +537,51 @@ def current_window(self):
     def add_strut(self, strut):
         from libqtile.bar import Bar, Gap
 
+        try:
+            if strut[0]:
+                x = strut[0]
+            elif strut[1]:
+                x = self.x_screen_dimensions.get('x') - strut[1]
+            else:
+                x = ((strut[8] + strut[9]) / 2 or (strut[10] + strut[11]) / 2)
+            if strut[2]:
+                y = strut[2]
+            elif strut[3]:
+                y = self.x_screen_dimensions.get('y') - strut[2]
+            else:
+                y = ((strut[4] + strut[5]) / 2 or (strut[6] + strut[7]) / 2)
+            screen = self.find_screen(x, y)
+            assert (screen is not None)
+        except AssertionError:  # strut Values are defined but the coordinates are not to be found on any screen
+            logger.error("Relevant screen could not be found")
+            return
```

Comment:
Maybe it is better to use a normal 
```py
if screen is None:
    logger.error("...")
    return
```
because it is shorter and also works with `python`s `-O` flag: [`assert` statements have no effect when running with `-O`](https://docs.python.org/3/reference/simple_stmts.html?highlight=assert#the-assert-statement). I don't know why python does it, but apparently it does.

Reply:
Forget that and mark it as resolved...

Reply:
I don't know, seems like a reasonable fix to me. That lets us drop a whole level of indentation too...